tests/access.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { test, expect } from '@playwright/test';
+import login, { logout } from '@lib/auth';
+import { config } from '@lib/config';
+import {
+  navigateToKafkaList,
+  deleteKafkaInstance,
+  createKafkaInstance,
+  navigateToAccess,
+  grantManageAccess,
+  findAccessRow,
+  revokeAccess
+} from '@lib/kafka';
+
+const testInstanceName = config.instanceName;
+
+test.beforeEach(async ({ page }) => {
+  await login(page);
+
+  await navigateToKafkaList(page);
+
+  await expect(page.getByRole('button', { name: 'Create Kafka instance' })).toBeVisible();
+
+  if ((await page.getByText(testInstanceName).count()) > 0 && (await page.locator('tr').count()) === 2) {
+    // Test instance present, nothing to do!
+  } else {
+    await page.waitForSelector('[role=progressbar]', {
+      state: 'detached',
+      timeout: config.kafkaInstanceCreationTimeout
+    });
+
+    for (const el of await page.locator(`tr >> a`).elementHandles()) {
+      const name = await el.textContent();
+
+      if (name !== testInstanceName) {
+        await deleteKafkaInstance(page, name);
+      }
+    }
+
+    if ((await page.getByText(testInstanceName).count()) === 0) {
+      await createKafkaInstance(page, testInstanceName);
+    }
+  }
+});
+
+// This test needs to run as an org admin until the new UI with refactored access dialog is released.
+test('test kafka manage access permission', async ({ page }) => {
+  await navigateToAccess(page, testInstanceName);
+  await grantManageAccess(page, config.username_secondary);
+
+  const row = await findAccessRow(page, config.username_secondary, '', 'Kafka Instance');
+  await expect(row).toHaveCount(1);
+
+  await logout(page);
+  await login(page, config.username_secondary, config.password_secondary);
+
+  await navigateToAccess(page, testInstanceName);
+  await grantManageAccess(page, 'All accounts');
+
+  const rowAllAccounts = await findAccessRow(page, 'All Accounts', 'Alter', 'Kafka Instance');
+  await expect(rowAllAccounts).toHaveCount(1);
+
+  await revokeAccess(page, 'All Accounts', 'Alter', 'Kafka Instance');
+  await revokeAccess(page, config.username_secondary, 'Alter', 'Kafka Instance');
```

Comment:
Shouldn't be there some kind of assert?

Reply:
I was thinking about this more like a test teardown cleanup, but yeah, will add something.

Reply:
Added assert directly to the function.